fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/rewrite/PlanTreeRewriteJob.java

```diff
@@ -48,7 +48,7 @@ protected RewriteResult rewrite(Plan plan, List<Rule> rules, RewriteJobContext r
                 Preconditions.checkState(newPlans.size() == 1,
                         "Rewrite rule should generate one plan: " + rule.getRuleType());
                 Plan newPlan = newPlans.get(0);
-                if (!newPlan.deepEquals(plan)) {
+                if (!newPlan.deepEquals(plan) || !newPlan.getLogicalProperties().equals(plan.getLogicalProperties())) {
```

Comment:
In my opinion, all Plan.equals() should be implemented to compare their own members and LogicalProperties.
If there is some thing wrong, maybe some Plan not compare the LogicalProperties.

This change is not quite correct, because it just compare the top plan's logicalProperties, child plan with different logicalProperties maybe meet same problem.